### DISCUSSION TITLE
Ultralytics v8.2.39 problem

### DISCUSSION THREAD

**Discussion by mo-atiff** [ultralytics/ultralytics#13887] [Q&A]
Just updated ultralytics to v8.2.39 and ran into errors while training a simple custom object detection model : 

`
from ultralytics import YOLO
import cv2

model = YOLO("yolov8n.pt")

results = model.train(data = "data.yaml", epochs=10, resume=True, device='cpu') # I'm using absolute path for data parameter
`

ERROR : 
SyntaxError: 'v5loader' is not a valid YOLO argument.
'fl_gamma' is not a valid YOLO argument.
'image_weights' is not a valid YOLO argument.

I do also get such warnings : 

WARNING ⚠️ 'hide_labels' is deprecated and will be removed in 'ultralytics 8.399999999999999' in the future. Please use 'show_labels' instead.
WARNING ⚠️ 'hide_conf' is deprecated and will be removed in 'ultralytics 8.399999999999999' in the future. Please use 'show_conf' instead.
WARNING ⚠️ 'line_thickness' is deprecated and will be removed in 'ultralytics 8.399999999999999' in the future. Please use 'line_width' instead.

**Comment by glenn-jocher** [comment]
@mo-atiff hi there! 👋

Thank you for reaching out and providing the details of your issue. Let's address this step-by-step:

1. **Reproducible Code Example**:
   Your provided code snippet is helpful, but to fully investigate, we need to ensure that the issue can be reproduced. Could you please confirm that the error persists with the latest versions of `torch` and `ultralytics`? If not, please upgrade your packages and try again:
   ```bash
   pip install --upgrade torch ultralytics
   ```

2. **Deprecation Warnings**:
   The warnings you are seeing indicate that certain arguments have been deprecated and will be removed in future versions. Here are the replacements:
   - `hide_labels` -> `show_labels`
   - `hide_conf` -> `show_conf`
   - `line_thickness` -> `line_width`

   Please update your code accordingly to avoid these warnings.

3. **Invalid Arguments**:
   The errors you're encountering (`'v5loader'`, `'fl_gamma'`, `'image_weights'` not valid YOLO arguments) suggest that there might be some legacy or incorrect arguments being passed. Ensure that your `data.yaml` file and any other configurations are up-to-date and compatible with YOLOv8.

4. **Example Code**:
   Here’s a refined version of your code snippet:
   ```python
   from ultralytics import YOLO
   import cv2

   model = YOLO("yolov8n.pt")

   results = model.train(data="data.yaml", epochs=10, device='cpu')  # Removed 'resume=True' for initial training
   ```

If the issue persists after these steps, please provide a minimum reproducible example that includes your `data.yaml` configuration. This will help us diagnose the problem more effectively. You can find more details on creating a minimum reproducible example [here](https://docs.ultralytics.com/help/minimum_reproducible_example).

Looking forward to your update! 😊

**Reply by glenn-jocher** [reply]
Fixed in https://github.com/ultralytics/ultralytics/pull/14091

**Comment by mo-atiff** [comment]
hey @glenn-jocher 

The same code works now no issues, I actually trained the v8 model on coco.yaml dataset which got completed with no issues, then I tried the same script with data.yaml, Interestingly it worked no such errors.

I actually trained v8 model yesterday but then today I saw a an upgrade of ultralytics and did upgrade(v8.2.39) soon after which I had ran into such issues ! 

I did remove the "device" parameter which doesn't matter ig but anyway I did

Thanks for your reply really appreciate it

**Reply by glenn-jocher** [reply]
Hello @mo-atiff,

Thank you for the update! I'm glad to hear that your issue has been resolved and that the training is working fine now. It’s great to know that you successfully trained the model on both the COCO dataset and your custom dataset.

Regarding the upgrade to v8.2.39 and the subsequent errors you encountered, it’s possible that there were some temporary inconsistencies or deprecated parameters that caused the issue. Removing the "device" parameter might have helped, but it’s also worth ensuring that your environment is fully updated.

For future reference, if you encounter similar issues, please make sure to:
1. **Check for the latest versions**: Ensure you are using the most recent versions of `torch` and `ultralytics`.
2. **Review deprecated warnings**: Pay attention to any deprecation warnings and update your code accordingly to avoid future issues.

If you run into any more problems or have further questions, feel free to reach out. The YOLO community and the Ultralytics team are always here to help!

Happy training! 🚀
